scylla/src/transport/connection_pool.rs

```diff
@@ -938,49 +941,94 @@ impl PoolRefiller {
         }
     }
 
+    #[cfg(feature = "cloud")]
+    fn maybe_translate_for_serverless(
+        &self,
+        mut endpoint: UntranslatedEndpoint,
+    ) -> impl Future<Output = UntranslatedEndpoint> {
+        if let Some(cloud_config) = self.pool_config.connection_config.cloud_config.clone() {
+            // If we operate in the Cloud, then we substitute every node's address with the proxy's of the datacenter that the node resides in.
+            futures::future::Either::Left(async move {
+                if let UntranslatedEndpoint::Peer(PeerEndpoint {
+                    host_id,
+                    ref mut address,
+                    ref datacenter,
+                    ..
+                }) = endpoint
+                {
+                    if let Some(dc) = datacenter.as_deref() {
+                        if let Some(dc_config) = cloud_config.datacenters.get(dc) {
+                            if let Ok(resolved) = resolve_hostname(&dc_config.server).await {
+                                *address = NodeAddr::RealPeer(resolved)
+                            } else {
+                                warn!( // FIXME: perhaps error! would fit here better?
+                                        "Couldn't resolve address: {} of datacenter {} that node {} resides in; node therefore skipped.",
```

Comment:
Doesn't "node therefore skipped" give a wrong message? This is just a failure of a single connection attempt. It will not cause the node to be "skipped" in any way.

Reply:
You're right; what content do you suggest for that message?